src/db/migrations/20180326231610_create-user_rankings-table.js

```diff
@@ -0,0 +1,24 @@
+const tableName = 'user_rankings'
+
+exports.up = knex => 
+  knex.schema.createTable(tableName, table => {
+    table.increments('id')
+    table.integer('item_id')
+      .notNullable()
+      .references('id')
+      .inTable('items')
+    table.integer('user_id')
+      .notNullable()
+      .references('id')
+      .inTable('users')
+    table.integer('prev_ranking_id')
+    table.timestamps(true, true)
+    table.timestamp('deleted_at')
+
+    table.foreign('prev_ranking_id')
+      .references('id')
+      .inTable('user_rankings')
+  })
```

Comment:
this blows my mind

Reply:
![](https://media.giphy.com/media/aKtApzd6XeN44/giphy.gif)